File: api/src/main/java/com/trabalho/api/dto/EnderecoDTO.java
package com.trabalho.api.dto;

import com.trabalho.api.model.Endereco;

import lombok.Data;

@Data
public class EnderecoDTO {
    private Long id;
    private String cidade;
    private String uf;
    private String bairro;
    private String rua;
    private String numero;
    private String complemento;
    private String latitude;
    private String longitude;

    public static EnderecoDTO convert(Endereco endereco){
        EnderecoDTO enderecoDTO = new EnderecoDTO();
        enderecoDTO.setBairro(endereco.getBairro());
        enderecoDTO.setCidade(endereco.getCidade());
        enderecoDTO.setComplemento(endereco.getComplemento());
        enderecoDTO.setId(endereco.getId());
        enderecoDTO.setLatitude(endereco.getLatitude());
        enderecoDTO.setLongitude(endereco.getLongitude());
        enderecoDTO.setNumero(endereco.getNumero());
        enderecoDTO.setRua(endereco.getRua());
        enderecoDTO.setUf(endereco.getUf());
        return enderecoDTO;
    }
}

File: cliente-pooweb2/src/app/app-core/dto/EnderecoDto.ts
export class EnderecoDto{
    id: number;
    cidade: string;
    uf: string;
    bairro: string;
    rua: string;
    numero: string;
    complemento: string;
    latitude: string;
    longitude: string;

    constructor(
        id: number,
        cidade: string,
        uf: string,
        bairro: string,
        rua: string,
        numero: string,
        complemento: string,
        latitude: string,
        longitude: string
    ){
        this.id = id;
        this.cidade = cidade;
        this.uf = uf;
        this.bairro = bairro;
        this.rua = rua;
        this.numero = numero;
        this.complemento = complemento;
        this.latitude = latitude;
        this.longitude = longitude;
    }
}